Drivers/c/UartHwas/test/main.cpp

```diff
@@ -0,0 +1,430 @@
+    }
+
+    constexpr uint32_t EXPECTED_TX_EMPTY_VALUE = UART_HWAS_IxR_TXEMPTY_MASK;
+    const uint32_t ACTUAL_TX_EMPTY_VALUE =
+        uartReg->sr & UART_HWAS_IxR_TXEMPTY_MASK;
+    LONGS_EQUAL(EXPECTED_TX_EMPTY_VALUE, ACTUAL_TX_EMPTY_VALUE);
+
+    LONGS_EQUAL(EXPECTED_RX_VALUE, rxByte);
+  }
+
+  static inline void Verify_Rx(const UartHwas &uart,
+                               const uint8_t EXPECTED_RX_VALUE) {
+    const Uart_Registers *uartReg = (Uart_Registers *)uart.uartAddress;
+
+    //< wait for byte to be send
+    while (!(uartReg->sr & UART_HWAS_IxR_TXEMPTY_MASK)) {
+    }
+
+    //< wait for byte to be read
+    while (uartReg->sr & UART_HWAS_IxR_RXRDY_MASK) {
+    }
+
+    LONGS_EQUAL(EXPECTED_RX_VALUE, rxByte);
+  }
+
+  UartHwas_Config config;
+
+  void teardown() override {
+    Register_set_bits(uart.uartAddress + UART_HWAS_IDR_OFFSET,
+                      UART_HWAS_IxR_TXEMPTY_MASK | UART_HWAS_IxR_RXRDY_MASK);
+    Register_set_bits(uart.uartAddress + UART_HWAS_CR_OFFSET,
+                      UART_HWAS_CR_RSTRX_MASK | UART_HWAS_CR_RSTTX_MASK);
+    memset(&config, 0, sizeof(config));
+  }
+};
+
+/// \Given uninitialized UART
+/// \When UART0 is initialized with 9600 bps
+/// \Then proper configuration shall be read.
+TEST(UartHwas, uart0Baud9600) {
+  //< Setup
+  config.baudrate = UartHwas_Baudrate_9600;
+  config.id = UartHwas_Id_0;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UAbr U/// \When UART1 is initialized with 9600 bps
+/// \Then proper configuration shall be read.
+TEST(UartHwas, uart1Baud9600) {
+  //< Setup
+  config.baudrate = UartHwas_Baudrate_9600;
+  config.id = UartHwas_Id_1;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UART
+/// \When UART2 is initialized with 9600 bps
+/// \Then proper configuration shall be read.
+TEST(UartHwas, uart2Baud9600) {
+  //< Setup
+  config.baudrate = UartHwas_Baudrate_9600;
+  config.id = UartHwas_Id_2;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UART
+/// \When UART3 is initialized with 9600 bps
+/// \Then proper configuration shall be read.
+TEST(UartHwas, uart3Baud9600) {
+  //< Setup
+  config.baudrate = UartHwas_Baudrate_9600;
+  config.id = UartHwas_Id_3;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UART
+/// \When UART4 is initialized with 9600 bps
+/// \Then proper configuration shall be read.
+TEST(UartHwas, uart4Baud9600) {
+  //< Setup
+  config.baudrate = UartHwas_Baudrate_9600;
+  config.id = UartHwas_Id_4;
+
+  //< Exercise
+  UartHwas_init(&uart, &config);
+
+  //< Verify
+  VerifyInit(uart, config);
+}
+
+/// \Given uninitialized UART
+/// \When UART0 is initialized with 115200 bps
```

Comment:
UART0 vs UartHwas_Id_4. 
On a side note, this piece of code does not require exhaustive testing of all possible Uarts..... Testing the one target config, or the two possible target configs (the ones connected to the actual board) should be enough.

Reply:
Changed to loop in new commit